htdocs/avoloicustomtypes/class/api_avoloicustomtypes.class.php

```diff
@@ -233,4 +233,197 @@ public function tierstype($id) {
 			throw new RestException(405, "Problème lors de la récupération du type de tiers");
 		}
 	}
+
+	/////////////// PARTIE TYPE DE RENDEZ-VOUS ///////////////
+
+	/**
+	 * Create an event type
+	 * 
+	 * @param 	string    $eventType		Event type
+	 * @param		string		$color_code		Color of the event type
+	 * @return  string									Created type ID
+	 *
+	 * @throws 500
+	 * @throws 501
+	 * @throws 400
+	 * @throws 401
+	 * @throws 404
+	 * @throws 200
+	 *
+	 * @url POST /createeventtype
+	 */
+	public function createeventtype($eventType, $color_code = "FFC266") {
+		global $conf, $langs, $user;
+
+		$libelle = $eventType;
+
+		// Création du code
+		$code = $this->generaterandomcode();		
+
+		// Incrémentation de l'id
+		$sqlId = "SELECT MAX(id) as id FROM `llx_c_actioncomm`";
+		$resqId = $this->db->query($sqlId);
+		$idTmp = $this->db->fetch_object($resqId);
+		$id = $idTmp->id + 1;
+
+		// Incrémentation de la position
+		$sqlPos = "SELECT MAX(position) as position FROM `llx_c_actioncomm`";
+		$resqlPos = $this->db->query($sqlPos);
+		$posTmp = $this->db->fetch_object($resql);
+		$position = $posTmp->position + 1;
+
+		$sql = "INSERT INTO `llx_c_actioncomm` (`id`, `code`, `type`, `libelle`, `active`, `color`, `position`)";
+		$sql.= " VALUES (";
+		$sql.= "'". $id ."', ";
+		$sql.= "'". $code ."', ";
+		$sql.= "'user', ";
+		$sql.= "'". $libelle ."', ";
+		$sql.= "1, ";
+		$sql.= "'". $color_code ."', ";
+		$sql.= "'". $position ."');";
+
+		$resql = $this->db->query($sql);
+
+		if ($resql) {
+			$sqlId = "SELECT MAX(id) as id FROM `llx_c_actioncomm`";
+			$rtdId = $this->db->query($sqlId);
+			$rtd = $this->db->fetch_object($rtdId);
+
+			return $rtd->id;
+		} else {
+			throw new RestException(405, "Problème lors de la création du type de rendez-vous");
+		}
+	}
+
+	/**
+	 * Update an event type
+	 * 
+	 * @param		string		$id						Event ID
+	 * @param 	string    $eventType		Event type
+	 * @param		string		$color_code		Color of the event type
+	 * @return  string									Created type ID
+	 *
+	 * @throws 500
+	 * @throws 501
+	 * @throws 400
+	 * @throws 401
+	 * @throws 404
+	 * @throws 200
+	 *
+	 * @url PUT /updateeventtype
+	 */
+	public function updateeventtype($id, $eventType, $color_code = "FFC266") {
+		global $conf, $langs, $user;
+
+		$libelle = $eventType;
+		
+		$sql = "UPDATE `llx_c_actioncomm` SET ";
+		$sql.= "`color`='$color_code', ";
+		$sql.= "`libelle`='$libelle' ";
+		$sql.= "WHERE id = $id";
+
+		$resql = $this->db->query($sql);
+
+		// Vérification de la bonne modification de la ligne
+		$sqlTmp = "SELECT * FROM `llx_c_actioncomm` WHERE id = $id";
+		$resqTmp = $this->db->query($sqlTmp);
+		$objTmp = $this->db->fetch_object($resqTmp);
+
+		if ($objTmp->libelle !== $libelle) {
+			throw new RestException(405, "Problème lors de la modification du type de rendez-vous");
```

Comment:
405 signifie Method not allowed, dans le cas present est plus dans un autre type de problème. 
Une 500 serait plus de circonstance je pense